flow360/component/simulation/validation/validation_simulation_params.py

```diff
@@ -264,3 +267,55 @@ def _validate_cht_has_heat_transfer(params):
                     )
 
     return params
+
+
+def _check_complete_boundary_condition_and_unknown_surface(params):
+    ## Step 1: Get all boundaries patches from asset cache
+    asset_boundary_entities = params.private_attribute_asset_cache.boundaries
+    if asset_boundary_entities is None or asset_boundary_entities == []:
+        return params
+    asset_boundaries = {boundary.name for boundary in asset_boundary_entities}
```

Comment:
```suggestion
    asset_boundaries = set(asset_boundary_entities)
```

Reply:
Did not commit this suggestion in the end because `Surface` obj is not hashable